interview/data_structure/singly_linked_list.py

```diff
@@ -0,0 +1,101 @@
+class Node:
+    """Single element of linked list. Each node has two different fields, *'data'*
+    containing the value to be stored in the node and *'next'* containing a reference
+    to the next element in the list.
+
+    :param str/int/float data: value of the node.
+    """
+
+    def __init__(self, data):
+        self.data = data
+        self.next = None
+
+    def __repr__(self):
+        return self.data
+
+
+class SinglyLinkedList:
+    """Linked list as a collection of nodes. First element of the linked list is called
+    '*head*'.
+    """
+
+    def __init__(self):
+        self.head = None
+
+    def create(self, data=None):
+        """Create linked list with some elements.
+
+        :param list data: initial element(s) to create linked list.
+        """
+        if data is not None:
+            nodes = data.copy()
+            node = Node(nodes.pop(0))
+            self.head = node
+            for n in nodes:
+                node.next = Node(n)
+                node = node.next
+
+    def add(self, data, position=None):
```

Comment:
Not sure whether it is more intuitive to break `add` into `insert` and `append`.

Reply:
I like it

Reply:
Done